### PR TITLE
invoke `cargo` to install mdbook

### PR DESCRIPTION
**Pull request by nikomatsakis**
cc @Michael-F-Bryan -- does this look right? 

(I'm also not sure what the `~/.cargo/env` command does, but it seemed likely it might be needed to run cargo, so I moved it up)

### CODE REVIEW

**Comment by nikomatsakis** [comment]
cc https://github.com/rust-lang-nursery/rustc-guide/issues/1

File: ci/install.sh [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 set -ex
 
+cargo install cargo install mdbook --vers "0.0.28"
```
[the remark this answers is not in the document]
Comment (nikomatsakis) [reply ci/install.sh]: Uh, yeah =)

[the remark this answers is not in the document]
Comment (nikomatsakis) [reply ci/install.sh]: OK :)

**Comment by nikomatsakis** [comment]
@Michael-F-Bryan take another look, if you please

**Comment by Michael-F-Bryan** [comment]
It looks good from what I can see, although the only real way to find out is to merge this PR into `master` and trigger uploading the rendered document.

To tell GitHub that Travis is allowed to push to GitHub Pages we'll also need to add an [API key][1] (with `public_repo` permissions) and then add the [encrypted] key to `travis.yml`.

[1]: https://github.com/settings/tokens
[encrypted]: https://docs.travis-ci.com/user/encryption-keys/

**Comment by nikomatsakis** [comment]
@Michael-F-Bryan ok; where do I put the key in travis.yml exactly? It is under `deploy`, as I [see here](https://docs.travis-ci.com/user/deployment/pages/)?

**Comment by nikomatsakis** [comment]
I feel like `mdbook` should have step-by-step instructions for this =)

**Comment by Michael-F-Bryan** [comment]
What I usually do is run `travis encrypt GH_TOKEN=$TOKEN --add` on the command-line and it'll add the token to `travis.yml` for you. It should end up looking something like [this].

> I feel like mdbook should have step-by-step instructions for this =)

I was thinking the exact same thing yesterday! Now that alternate backends have been merged into `mdbook`, I was thinking of making a backend which will automatically build your book and then push it up to GitHub Pages for you, printing detailed messages if anything is wrong with your environment. As part of that tool you'd mention how to set everything up in the README.

I've always found CI to be a massive pain to get set up, adding a bunch of noise to the git history because you'll typically need to make a commit to `master` push it up, wait 5-10 minutes for the build to fail, make a one-line change to see if you've fixed the issue, rinse and repeat. Appveyor is even worse because it only runs one build at a time for your entire project :disappointed: 

[this]: https://github.com/rust-lang-nursery/mdBook/blob/master/.travis.yml#L16

**Comment by nikomatsakis** [comment]
OK, well, I'm going to merge this. We'll see what happens I suppose.

**Pull request merged**